### PR TITLE
Merge LandingPage Streamfields

### PR DESCRIPTION
**Pull request by kurtw**
- Merge LandingPage Streamfields
- Update template with new StreamFields
- Add lambda function into the context that returns fake related posts data for prototyping
- Fix small error in Text Introduction prototype data

NOTE: This pulled in the commit from #1223 to work. 
@kave 
@jimmynotjim

### CODE REVIEW

File: cfgov/v1/models/organisms.py [hunk 1]
```diff
@@ -66,9 +66,16 @@ class Meta:
...
 class RelatedPosts(blocks.StructBlock):
     limit = blocks.CharBlock(default='3', label='Limit')
-    relate_posts = blocks.BooleanBlock(required=False, default=True, label='Blog Posts')
-    relate_newsroom = blocks.BooleanBlock(required=False, default=True, label='Newsroom')
-    relate_events = blocks.BooleanBlock(required=False, default=True, label='Events')
+    show_heading = blocks.BooleanBlock(required=False, default=True,
+                                       label='Show Heading and Icon?',
+                                       help_text='This toggles the heading and'
+                                       + ' icon for the related types.')
+    relate_posts = blocks.BooleanBlock(required=False, default=True,
```
Comment (kave) [inline cfgov/v1/models/organisms.py]: lets either remove newsroom & blog checkboxes or added `editable=False` to disable it since this now works querying on tags

File: cfgov/jinja2/v1/landing-page/index.html [hunk 2]
```diff
@@ -1,22 +1,27 @@
-
 {% extends 'layout-2-1-bleedbar.html' %}
 
 {% import 'molecules/related-posts.html' as related_posts with context %}
 
 {% if not page %}
-      {% set doc = get_document('posts', 'four-years-working-for-you') %}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Why did you remove the indentation? It's how we've done every other prototyped data block.

File: cfgov/jinja2/v1/landing-page/index.html [hunk 3]
```diff
@@ -26,93 +31,48 @@
...
-                 block__flush-top">
-            {% if prototype %}
-                {% include 'molecules/text-introduction.html' %}
-            {% endif %}
-            {% for block in page.introduction %}
-                {{ render_stream_child(block) }}
-            {% endfor %}
-        </div>
-    {% endif %}
+    {% if prototype %}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: We know this is a prototype, we set the value within the same file. Is this conditional necessary?

File: cfgov/jinja2/v1/landing-page/index.html (hunk 2 above)
Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: I've seen it done in multiple ways. I can reindent.

File: cfgov/jinja2/v1/landing-page/index.html (hunk 3 above)
Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: We don't know if it's a prototype. It's only a prototype if the value is not provided in the context, hence the `{% if value %}` block.

**Comment by kurtw** [comment]
Updated with slight changes. Prototype page is at http://localhost:8000/landing-page/

File: cfgov/jinja2/v1/landing-page/index.html (hunk 2 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Mostly it makes reviewing harder because it's a big diff for little change, if any change.

File: cfgov/jinja2/v1/landing-page/index.html (hunk 3 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Ha, sorry, I didn't realize this was serving for both prototype and CMS.

File: cfgov/jinja2/v1/landing-page/index.html [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
             'page':  {
                 'seo_title': 'This is a prototype page for testing landing page layouts.',
-                'related_posts': {
-                    'posts': more_like_this(doc, search_types='posts', search_size=3),
-                    }
+                'related_posts': global_dict.related_posts_function
+            },
+            'related_posts_settings': {
+                'value': {
+                    'limit': 3,
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Left align values here and throughout.

Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: Could you clarify?

File: cfgov/jinja2/v1/landing-page/index.html [hunk 5]
```diff
@@ -26,93 +32,48 @@
...
-                {% for x in range(0, 3) %}
-                    {% include 'molecules/image-text-25-75.html' %}
-                {% endfor %}
-            {% endif %}
-            {% for block in page.image_text_25_75_content %}
-                {{ render_stream_child(block) }}
-            {% endfor %}
-        </div>
-    {% endif %}
+        <h2>Image Text 25 75 Group Header</h2>
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Should the 25/75 still have a block wrapping it?

Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: I was told that the block classes aren't needed anymore...

File: cfgov/jinja2/v1/landing-page/index.html [hunk 6]
```diff
@@ -26,93 +32,48 @@
...
-            {% if prototype %}
-                {% include 'molecules/text-introduction.html' %}
-            {% endif %}
-            {% for block in page.introduction %}
-                {{ render_stream_child(block) }}
-            {% endfor %}
-        </div>
-    {% endif %}
+    {% if prototype %}
+        {% include 'molecules/text-introduction.html' %}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: The text introduction should for be wrapped in a `block`. If not here, we need to update the macro

Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: I'd re-add all the `block` wrappers that were removed to fix the layout, then we can address whether they should be in the templates or the molecules/organisms in another PR.

File: cfgov/jinja2/v1/sublanding-page/index.html [hunk 7]
```diff
@@ -24,14 +24,9 @@
...
-           and 'hero' not in block.block_type %}
-            {% if 'heading' in block.block_type %}
-                <h2>{{ block }}</h2>
-            {% else %}
-                {{ render_stream_child(block) }}
-            {% endif %}
-        {% endif %}
+        <div class="block">
+            {{ render_stream_child(block) }}
+        </div>
```
Comment (kave) [inline cfgov/jinja2/v1/sublanding-page/index.html]: I'm guessing this hero logic is no longer a factor?

File: cfgov/jinja2/v1/landing-page/index.html (hunk 4 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: We've been indenting the values after the keys so that it's easier to scan visually. See https://github.com/cfpb/cfgov-refresh/blob/flapjack/cfgov/jinja2/v1/_includes/molecules/form-field-with-button.html#L13-L24

File: cfgov/jinja2/v1/sublanding-page/index.html (hunk 7 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/sublanding-page/index.html]: We discussed in scrum that it'll be up to content creators to ensure they don't add both, and editors to review the posts before publishing to catch any slip ups.

File: cfgov/jinja2/v1/_includes/organisms/image-text-25-75-group.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<div class="o-image-text-25-75-group">
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/organisms/image-text-25-75-group.html]: I like it

File: cfgov/jinja2/v1/landing-page/index.html [hunk 9]
```diff
@@ -26,105 +32,61 @@
...
-        <div class="block
-                 block__flush-top">
-            {% if prototype %}
-                {% include 'molecules/text-introduction.html' %}
-            {% endif %}
-            {% for block in page.introduction %}
-                {{ render_stream_child(block) }}
-            {% endfor %}
+    {% if prototype %}
+        <div class="block">
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Checking locally, it looks like this block still needs the `flush-top` modifier

File: cfgov/jinja2/v1/landing-page/index.html [hunk 10]
```diff
@@ -26,105 +32,61 @@
...
+            <div class="o-link-blob-group">
+                <h2>Link Blob Header</h2>
+                {% include 'molecules/half-width-link-blob.html' %}
+                {% include 'molecules/half-width-link-blob.html' %}
+            </div>
         </div>
+    {% else %}
+        {% for block in page.header -%}
+            <div class="block">
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: This is an extra block, the one on L:78 is sufficient

File: cfgov/jinja2/v1/landing-page/index.html (hunk 4 above)
Comment (kurtw) [inline cfgov/jinja2/v1/landing-page/index.html]: The values are indented...? `limit` is indented 4 spaces from `value` which is indented 4 spaces from `related_posts_settings`

**Comment by kave** [comment]
tested looks good to me :+1:

File: cfgov/jinja2/v1/landing-page/index.html (hunk 4 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/landing-page/index.html]: Sorry, in the `key: value` pair, we're indenting the value side. Dual meanings of the same word, my bad.

**Comment by jimmynotjim** [comment]
:+1:
